### PR TITLE
VIH-9822 remove menus, replace with select component

### PR DESCRIPTION
**Pull request by phillipjchambers**
### JIRA link (if applicable) ###
VIH-9822

### Change description ###
* replace 'menu' components with a single 'select' component in Allocate Hearings
* fix tests

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/allocate-hearings/allocate-hearings.component.spec.ts [hunk 1]
```diff
@@ -54,27 +56,37 @@ describe('AllocateHearingsComponent', () => {
...
 
     beforeEach(() => {
         fixture = TestBed.createComponent(AllocateHearingsComponent);
         component = fixture.componentInstance;
         fixture.detectChanges();
-        component.csoMenu = TestBed.createComponent(JusticeUserMenuStubComponent).componentInstance as JusticeUsersMenuComponent;
-        component.caseTypeMenu = TestBed.createComponent(CaseTypeMenuStubComponent).componentInstance as CaseTypesMenuComponent;
-        component.csoAllocatedMenu = TestBed.createComponent(JusticeUserMenuStubComponent).componentInstance as JusticeUsersMenuComponent;
-        component.csoFilterMenu = TestBed.createComponent(JusticeUserMenuStubComponent).componentInstance as JusticeUsersMenuComponent;
+        // component.csoMenu = TestBed.createComponent(JusticeUserMenuStubComponent).componentInstance as JusticeUsersMenuComponent;
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/allocate-hearings/allocate-hearings.component.spec.ts:78]: could we remove the commented out code?

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/allocate-hearings/allocate-hearings.component.ts [hunk 2]
```diff
@@ -206,19 +244,23 @@ export class AllocateHearingsComponent implements OnInit {
...
-            const csoUsername = this.csoAllocatedMenu?.selectedLabel;
-            // safe to cast to string
-            const csoId = this.csoAllocatedMenu?.selectedItems as string;
-
             this.allocationHearingViewModel.checkHearing(hearing_id);
-            this.attemptToAssignCsoToSelectedHearings(csoId, csoUsername);
+
+            const selectedCso = this.selectAllocateCso?.selected as SelectOption;
+            if (selectedCso) {
+                console.log('selectedCso', selectedCso);
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/allocate-hearings/allocate-hearings.component.ts:258]: can probably remove this, or alternatively use `Logger` instead

**Review by oliver-scott — COMMENTED** [review]
looks good, just a couple of comments. Maybe some additional test coverage that can be added from looking at the SonarCloud results, but not a deal-breaker

**Comment by shaed-parkar** [comment]
> looks good, just a couple of comments. Maybe some additional test coverage that can be added from looking at the SonarCloud results, but not a deal-breaker

I would probably argue some of the negative scenarios (at least when no option is selected for justice user) is covered to ensure handling common scenarios

File: AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/allocate-hearings/allocate-hearings.component.ts (hunk 2 above)
Comment (phillipjchambers) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/allocate-hearings/allocate-hearings.component.ts:258]: Done

**Review by oliver-scott — APPROVED** [review]

**Review by shaed-parkar — DISMISSED** [review]

**Review by oliver-scott — APPROVED** [review]

**Review by shaed-parkar — APPROVED** [review]

**Comment by komaldabhi1509** [comment]
Approved

**Review by komaldabhi1509 — APPROVED** [review]
Approved

**Pull request merged**